boa_parser/src/lexer/token.rs

```diff
@@ -95,7 +95,7 @@ impl From<BigInt> for Numeric {
 #[derive(Clone, PartialEq, Debug)]
 pub enum TokenKind {
     /// A boolean literal, which is either `true` or `false`.
-    BooleanLiteral(bool),
+    BooleanLiteral((bool, ContainsEscapeSequence)),
```

Comment:
```suggestion
    BooleanLiteral(bool),
```

I think it's best to leave it as is, and throw the error when we create the `TokenKind::BooleanLiteral` (same for `TokenKind::NullLiteral`), since a boolean and null literal shouldn't have escape sequence.

Reply:
Thanks for the suggestion.

But, if I throw an error when `Token::BooleanLiteral` is created at `boa_parser/src/lexer/identifier.rs`, it will reject the following code too:
```
let o = { f\u{61}lse: 10}; // should work
o.f\u{61}lse
```

Shouldn't we throw an error only when a `TokenKind::BooleanLiteral` or `TokenKind::NullLiteral` is parsed as a literal?

Reply:
ah, I see, You are right! Forgot about those cases :sweat_smile: